plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1763,125 @@ public void testOptimizeUsingForcedPartitioning()
         assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(union(initialFiles, updatedFiles));
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+            throws Exception
+    {
+        String tableName = "test_register_table_with_table_location_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+        // Wait for TEST_METADATA_CACHE_TTL_SECONDS + 1 seconds to get the metastore cache invalidated
+        SECONDS.sleep(TEST_METADATA_CACHE_TTL_SECONDS + 1);
```

Comment:
I am thinking to use `CachingHiveMetastore#invalidateTable` here

Reply:
Can we simply register it under a new name?

Reply:
There is no issue registering tables with the same name or different names. Because the metadata cache gets invalidated in `metastore.createTable` call.

Here, we are trying to ensure the table isn't accessible using `SELECT` query after dropping table from metastore.

See @ebyhr's comment https://github.com/trinodb/trino/pull/14779#discussion_r1023531300

Reply:
Thanks for the pointer.

Change the comment to `// Verify that dropTableFromMetastore actually works` and use invalidate instead of the sleep.

Reply:
I have created `metastore` of `CachingHiveMetastore` type. But somehow `metastore.invalidateTable` does not invalidate the cache. Not sure why?

```        
        this.metastore = cachingHiveMetastore(
            new BridgingHiveMetastore(testingThriftHiveMetastoreBuilder()
                    .metastoreClient(hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
                    .build()),
                newCachedThreadPool(daemonThreadsNamed("delta-%s")),
            new Duration(1, SECONDS),
            Optional.of(new Duration(1, SECONDS)),
            10000,
            false);

Reply:
Created a new `queryRunner` without cache and verified that table does not exist. Removed sleep.

Reply:
I added `flush_metadata_cache` call  instead of a new query runner.